src/fspec-archive.c

```diff
@@ -41,11 +41,26 @@ filetype(const char *type)
     exit(1);
 }
 
+static int
+defaultmode(const char *type)
+{
+    if (strcmp(type, "reg") == 0)
+        return 0644;
+    if (strcmp(type, "dir") == 0)
+        return 0755;
+    if (strcmp(type, "sym") == 0)
+        return 0777;
+    fprintf(stderr, "unknown file type '%s'\n", type);
+    exit(1);
+}
+
 int main(int argc, char **argv)
 {
-    char *line = NULL;
-    size_t len = 0;
```

Comment:
We actually need these initializations for the first call to getline (so it knows nothing was previously allocated).

Reply:
Fuck Im sorry, what a fail.

Reply:
No worries, that's what reviews are for.